src/plugins/pbjx/components/history/delegate.js

```diff
@@ -1,33 +1,68 @@
 import clearResponse from '@triniti/cms/plugins/pbjx/actions/clearResponse';
+import dirtyEditor from '@triniti/cms/plugins/blocksmith/actions/dirtyEditor';
+import storeEditor from '@triniti/cms/plugins/blocksmith/actions/storeEditor';
+import convertToEditorState from '@triniti/cms/plugins/blocksmith/utils/convertToEditorState';
+import Message from '@gdbots/pbj/Message';
+import DateTimeType from '@gdbots/pbj/types/DateTimeType';
+import { change } from 'redux-form';
+import camelCase from 'lodash/camelCase';
+import StreamId from '@gdbots/schemas/gdbots/pbjx/StreamId';
 
-export default (dispatch) => ({
+
+export default (dispatch, ownProps) => ({
   /**
    * Initializes the container.
    *
    * This is needed because state is not wiped out
    * when components/container are unmounted.
-   *
-   * @param {StreamId} streamId - An instance of StreamId
-   * @param {Schema} schema - An instance of Schema to use to create requests.
    */
-  handleInitialize: (streamId, schema) => {
+  handleInitialize: () => {
+    const { node, schema } = ownProps;
+    const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
     dispatch(schema.createMessage().set('stream_id', streamId));
   },
 
   /**
    * Handles load more events.
    *
-   * @param {StreamId} streamId - An instance of StreamId
-   * @param {Schema} schema - An instance of Schema to use to create requests.
    * @param {string} since - Return events since this time.
    */
-  handleLoadMore: (streamId, schema, since) => {
+  handleLoadMore: (since) => {
+    const { node, schema } = ownProps;
+    const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
     dispatch(schema.createMessage({
       stream_id: streamId,
       since,
     }));
   },
 
+  /**
+   * Handles Revert.
+   *
+   * @param {array} selected
+   */
+  handleRevert: (selected) => {
```

Comment:
@rsumilang having this pbjx component directly import and use blocksmith stuff is making me itchy. can you turn lines 51-54 into a blocksmith util `revertBlocks` (or `restoreBlocks` depending on where we land on that) that is then passed into the component. pbjx is low level, it would be better if it were completely unaware of blocksmith

Reply:
If I create `revertBlocks` as a blocksmith util, then the history component will need to import it and pass it into its props. Doesn't feel like this solution gets to rid of knowing about blocksmith completely and introduces that other weird thing of passing in a prop to a component that is now required but cannot be defined as required on the component that accepts it because it doesn't use it, the delegate does.

Reply:
Perhaps it would make more sense if the History component was its own plugin. But idk if I care that much.

Reply:
@rsumilang it would be imported by the forms and passed in. that way this pbjx component doesnt know about a blocksmith util, just a blocks util

the delegate/component required props is what it is, not that big of a deal we just do the best we can

Reply:
Scratch that comment about the history component importing it in. It would be every screens form and passing to the history component. However the delegate still requires thus making it have some knowledge of it unless it was some abstracted handler. I mean, yes pbjx feels like a lower level thing but it has knowledge of form controls and blocksmith feels like a form thing.